src/Morphir/TypeScript/AST.elm

```diff
@@ -31,22 +37,47 @@ type Privacy
     | Private
 
 
+{-| Represents the path to a module nameSpace, for creating import statements or references
+-}
+type alias NamespacePath =
+    { packagePath : Path
+    , modulePath : Path
+    }
+
+
+namespacePath : Path -> Path -> NamespacePath
+namespacePath packagePath modulePath =
+    { packagePath = packagePath
+    , modulePath = modulePath
+    }
+
+
 {-| Represents a type definition.
 -}
 type TypeDef
-    = TypeAlias
-        { name : String
+    = Namespace
+        { name : Name
+        , privacy : Privacy
+        , content : List TypeDef
+        }
+    | TypeAlias
+        { name : Name
         , doc : String
         , privacy : Privacy
         , variables : List TypeExp
         , typeExpression : TypeExp
         }
     | Interface
-        { name : String
+        { name : Name
         , privacy : Privacy
         , variables : List TypeExp
         , fields : ObjectExp
         }
+    | ImportStatement
+        { name : Name
+        , privacy : Privacy
+        , typeExpression : TypeExp
+        }
```

Comment:
as mentioned above, I think Namespace and ImportStatement should be made types of their own, instead of being variants of TypeDef.

Reply:
I agree that they're not "type definitions", but I do think that all four things belong in the same list. Perhaps it's the name "type definitions" that should change.

#### `ImportStatement` variant
First thing: I should probably be clear that the `ImportStatement` variant isn't for the standard imports at the top of the file. The import statements at the top of the TypeScript file aren't directly represented in the AST at all. They're all generated in the Pretty-Printer. The `ImportStatement` variant is specifically for the strange "`export import`" statements that appear in the top level namespace file. eg these statements from the [sample output](https://github.com/Cynical-Optimist/TypeScriptOutputSamples/blob/dd83ecb6880d53b06a862c5ce566f013af95cdcd/sample-output-morphir-ir/Morphir.ts#L23-L28): 

```
  export namespace IR {
    export import Distribution = Morphir_IR_Distribution
    import Documented = Morphir_IR_Documented
    export import FQName = Morphir_IR_FQName
  ...
```

I don't really understand the `export import` syntax being used here, but these statements are taking an imported namespace, changing the name, and (optionally) re-exporting them. They can appear in the main body of a file, and they can appear inside a namespace.

Perhaps we should change the name of this variant to make it clearer? How about `ExportableImport` or `ImportRename`, or `ImportAlias`?

#### Keeping `Namespace` and `ImportStatement` variants
To produce the current output, a module definition needs to contain a list of "named things". So far, those things could be interface declarations, type alias declarations, exportable import declarations, or namespaces.

That's a useful set of things to work with. For instance, it's the exact same set of "named things" that can appear inside a namespace, and the same set of things which may contain a reference to something outside of the module. That means it's the set of things that we need to be able to handle in the function `collectRefsFromTypeDef`.

So there's a few different places in the code where we want to handle the exact set `( Namespace | TypeAlias | Interface | ImportStatement )`. IMHO that makes it useful to collect them together as variants of one type.

Meanwhile, I'm not sure if there are any places in the code where we want to handle the specific set `( TypeAlias | Interface )`.

Reply:
Though I do realize: I realize I might be locked into seeing the code a certain way.

This set of AST types is a good fit for the code I've written, but maybe that's just because I wrote the code with that set of AST types in mind.

I'd be interested to see a different approach if you have one in mind.

Reply:
Actually, I do think we can remove `NamespaceRef` from the list of `TypeExp` variants. It's not serving much purpose, and it's definitely not a 'Type'.

I could rewrite the original `ImportStatement` variant to just directly contain the namespacePath.

```
    | ImportAlias
        { name : Name
        , privacy : Privacy
        , namespace : NamespacePath
        }
```

That would limit the `ImportAlias` variant to only ever rename imported namespaces, and prevent it from ever handling any other kind of expression.

As I said, I don't fully understand this `export import newName = oldName` syntax, so I'm not sure if it ought to be able to handle other types anyway.

Reply:
> First thing: I should probably be clear that the ImportStatement variant isn't for the standard imports at the top of the file. The import statements at the top of the TypeScript file aren't directly represented in the AST at all. They're all generated in the Pretty-Printer. 

Thanks for clarifying, I had misunderstood that.

> Perhaps we should change the name of this variant to make it clearer? How about ExportableImport or ImportRename, or ImportAlias?

I think `ImportAlias` is the clearest option. We'll only use this for re-importing whole modules, right? So your proposed rewrite to hold the NamespacePath directly makes sense to me.

Given those changes, I can accept having it part of the `TypeDef` variant.